packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -67,14 +81,41 @@ class CodeSizeController {
       // If not a leaf node, build all children then take the sum of the
       // children's sizes as its own size.
       for (dynamic child in rawChildren) {
-        final childTreemapNode = generateTree(child, showDiff: showDiff);
+        final childTreemapNode = generateTree(
+          child,
+          diffTreeType: diffTreeType,
+          showDiff: showDiff,
+        );
+        if (childTreemapNode == null) {
+          continue;
+        }
         treemapNodeChildren.add(childTreemapNode);
         treemapNodeSize += childTreemapNode.byteSize;
       }
     } else {
       // If a leaf node, just take its own size.
-      // Defaults to 0 if a leaf node has a size of null.
-      treemapNodeSize = treeJson['value'] ?? 0;
+      treemapNodeSize = treeJson['value'];
+
+      // Only add nodes with a size.
+      if (treemapNodeSize == null || treemapNodeSize == 0) {
```

Comment:
what if treemapNodeSize == 0 because of equal positive and negative diffs below? though maybe I'm misunderstanding what treemapNodeSize is.

Reply:
perhaps tweak the comment to say "Only add leaf nodes with a size" and add a TODO to figure out why we are getting null sizes